includes/apple-exporter/components/class-footnotes.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Publish to Apple News: \Apple_Exporter\Components\Footnotes class
+ *
+ * @package Apple_News
+ * @subpackage Apple_Exporter\Components
+ */
+
+namespace Apple_Exporter\Components;
+
+/**
+ * A translation of the WordPress Footnotes block.
+ *
+ * @since 2.5.0
+ */
+class Footnotes extends Component {
+
+	/**
+	 * Look for node matches for this component.
+	 *
+	 * @param \DOMElement $node The node to examine for matches.
+	 * @access public
+	 * @return \DOMElement|null The node on success, or null on no match.
+	 */
+	public static function node_matches( $node ) {
+		if (
+			'ol' === $node->nodeName &&
+			self::node_has_class( $node, 'wp-block-footnotes' )
+		) {
+			return $node;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Register all specs for the component.
+	 *
+	 * @access public
+	 */
+	public function register_specs() {
+		$this->register_spec(
+			'footnotes-json',
+			__( 'Footnotes JSON', 'apple-news' ),
+			[
+				'role'       => 'container',
+				'layout'     => 'body-layout',
+				'components' => '#components#',
+			]
+		);
+	}
+
+	/**
+	 * Build the component.
+	 *
+	 * @param string $html The HTML to parse into text for processing.
+	 * @access protected
+	 */
+	protected function build( $html ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		preg_match_all( '/<li.*?>.*?<\/li>/', $html, $matches );
```

Comment:
I know this plugin has regular expression usage everywhere for working with HTML, but would it be possible to use the [WP_HTML_Tag_Processor class](https://developer.wordpress.org/reference/classes/wp_html_tag_processor/) to do this work instead?

Reply:
@kevinfodness I can't find that you can get/modify the content or the tag with WP_HTML_Tag_Processor. I tried that first. I could only find that you could change attributes.